File: src/greet.test.ts
import { test, expect, describe } from "bun:test";
import { greet } from "./greet";

describe("greet function", () => {
  test("should greet with default greeting", () => {
    const result = greet("World");
    expect(result).toBe("Hello, World!");
  });

  test("should greet with custom greeting", () => {
    const result = greet("TypeScript", "Welcome");
    expect(result).toBe("Welcome, TypeScript!");
  });

  test("should handle empty name", () => {
    const result = greet("");
    expect(result).toBe("Hello, !");
  });

  test("should handle special characters in name", () => {
    const result = greet("José María");
    expect(result).toBe("Hello, José María!");
  });

  test("should handle custom greeting with special characters", () => {
    const result = greet("Developer", "¡Hola");
    expect(result).toBe("¡Hola, Developer!");
  });
});

File: src/test/java/Greet/GreetTest.java
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

public class GreetTest {

    @Test
    void shouldGreetWithDefaultGreeting() {
        String result = Greet.greet("World");
        assertThat(result).isEqualTo("Hello, World!");
    }

    @Test
    void shouldGreetWithCustomGreeting() {
        String result = Greet.greet("Java", "Welcome");
        assertThat(result).isEqualTo("Welcome, Java!");
    }

    @Test
    void shouldHandleEmptyName() {
        String result = Greet.greet("");
        assertThat(result).isEqualTo("Hello, !");
    }

    @Test
    void shouldHandleSpecialCharactersInName() {
        String result = Greet.greet("José María");
        assertThat(result).isEqualTo("Hello, José María!");
    }

    @Test
    void shouldHandleCustomGreetingWithSpecialCharacters() {
        String result = Greet.greet("Developer", "¡Hola");
        assertThat(result).isEqualTo("¡Hola, Developer!");
    }
}
